app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -503,8 +513,18 @@ def vm_instance_defaults
         :volumes    => [new_volume],
         :scsi_controller => { :type => scsi_controller_default_type },
         :datacenter => datacenter,
+        :firmware => 'automatic',
         :boot_order => ['network', 'disk']
       )
     end
+
+    def firmware_mapping(firmware)
+      case firmware
+      when /UEFI/
+        'efi'
+      else
+        'bios'
```

Comment:
One more note, looks like all cases falls down here, even non-Intel platforms. I doubt VmWare supports anything other than x86, you could make an extra check for architecture. But BIOS is a sane default I think, less code less bugs. +1